diesel_compile_tests/tests/compile-fail/array_expressions_must_be_correct_type.rs

```diff
@@ -0,0 +1,20 @@
+#[macro_use]
+extern crate diesel;
+
+use diesel::*;
+use diesel::dsl::*;
+
+fn main() {
+    let connection = PgConnection::establish("").unwrap();
+    select(array((1, 3))).get_result::<Vec<i32>>(&connection);
+    select(array((1f64, 3f64))).get_result::<Vec<i32>>(&connection);
+    //~^ ERROR E0277
```

Comment:
Can we be specific on at least one of these? (Also why are there so many?)

Reply:
They're like this:

```
unexpected errors (from JSON output): [
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:5: 10:11: the trait bound `f64: diesel::SelectableExpression<()>` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:33: 10:43: the trait bound `f64: diesel::expression::NonAggregate` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:33: 10:43: the trait bound `f64: diesel::SelectableExpression<()>` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:33: 10:43: the trait bound `f64: diesel::query_builder::QueryId` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:12: 10:17: the trait bound `f64: diesel::Expression` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:33: 10:43: the trait bound `f64: diesel::query_builder::QueryFragment<_>` is not satisfied [E0277]"
    },
    Error {
        line_num: 10,
        kind: Some(
            Error
        ),
        msg: "10:12: 10:31: the trait bound `f64: diesel::Expression` is not satisfied [E0277]"
    }
]
```

I'll add some specifics.

Reply:
Ughhhhh actually add none of those. Seriously Rust. Just fucking say "`f64: AsExpression<Array<_>>` is not satisfied" these errors are ridiculous.

Reply:
Got it.